src/bcf/header.rs

```diff
@@ -35,21 +35,35 @@ pub struct Header {
 }
 
 impl Header {
-    /// Create a new header.
+    /// Create a new (empty) `Header`.
     pub fn new() -> Self {
         Header {
             inner: unsafe { htslib::bcf_hdr_init(ffi::CString::new(&b"w"[..]).unwrap().as_ptr()) },
             subset: None,
         }
     }
 
+    /// Create a new `Header` using the given `HeaderView as the template.
+    ///
+    /// After construction, you can modify the header independently from the template `header`.
+    ///
+    /// # Arguments
+    ///
+    /// - `header` - The `HeaderView` to use as the template.
     pub fn with_template(header: &HeaderView) -> Self {
```

Comment:
THis is a good opportunity to rename the function to `from_template`, following the Rust API guidelines for constructors.

Reply:
done